engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -152,6 +159,21 @@
     private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
     private ShadowMapResolutionDependentFBOs shadowMapResolutionDependentFBOs;
 
+    private static RenderGraph renderGraph = new RenderGraph(); // TODO: Try making this non-static
+
+    // Required for the reflection magic that Systems use.
```

Comment:
We discovered this together earlier when @vampcat was trying to make `WorldRendererImpl` a ComponentSystem - it failed to register entirely due to some init issue. I suspected it was from Gestalt wanting to use reflection, which goes wrong if there is no plain no-arg public constructor.

So the empty constructor essentially exists purely to let Gestalt register the system (via `@RegisterSystem`). It isn't actually used normally anywhere in code. But if you don't initialize the class variables you get errors from Java thinking there's a way to instantiate the class without fully initializing.

So it is a kind of ugly workaround right now to let commands work in that class via way of making it a System. I dunno if that's better or worse than trying to make it register commands differently, or even using other classes.

Also not sure if it is / should be a best practice that ES classes be structured in a way where they are never explicitly created from somewhere else (since Gestalt's angle is to auto-register stuff via reflection magic and annotations)